src/main/java/com/rsmart/kuali/kfs/cr/batch/CheckReconciliationImportStep.java

```diff
@@ -814,19 +842,50 @@ private CheckReconciliation setCheckReconciliationAttributes(Map<Integer, String
         return cr;
     }
     
-    
-   private GregorianCalendar getGregorianCalendar(String yyMMDD){
-	   String year  = "20"+yyMMDD.substring(0,2);
-	   String month = yyMMDD.substring(2,4);
-	   String day = yyMMDD.substring(4);
-	   
-	   month = ""+(Integer.parseInt(month) -1);
-	   
-	   GregorianCalendar gc = new GregorianCalendar(Integer.parseInt(year),Integer.parseInt(month),Integer.parseInt(day));
-	 
-	   return gc;
-	   
-   }
+    private Function<String, Date> getDateParser() {
+        String dateFormatString = getCrImportParameterValueAsString(CRConstants.CHECK_DATE_FORMAT);
+        if (StringUtils.equals(dateFormatString, CRConstants.LEGACY_DATE_FORMAT_yyMMdd)) {
+            return this::getDateFromStringWithTwoDigitYear;
+        } else {
+            final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern(dateFormatString, Locale.US);
+            return dateString -> getDateFromString(dateString, dateFormatter);
+        }
+    }
+
+    private Date getDateFromString(String dateString, DateTimeFormatter dateFormatter) {
+        LocalDate localDate;
+        if (StringUtils.isBlank(dateString) || ALL_ZEROS_PATTERN.matcher(dateString).matches()) {
+            LOG.warn("getDateFromString, Row has a blank or all-zero date; defaulting to 12/31/2099");
+            localDate = LocalDate.of(YEAR_2099, MONTH_12, DAY_31);
+        } else {
+            localDate = LocalDate.parse(dateString, dateFormatter);
+        }
+        Instant dateAsInstant = localDate.atStartOfDay(ZoneId.systemDefault())
+                .toInstant();
```

Comment:
I often use separate lines when chaining lots of method calls, such as when performing Stream operations. However, since there's very little chaining in this statement, I'll update it to be on one line.